Content.Server/_Shitmed/Medical/Surgery/SurgerySystem.cs

```diff
@@ -141,6 +148,113 @@ private void OnUtilityVerb(Entity<SurgeryToolComponent> ent, ref GetVerbsEvent<U
     private void OnSurgeryStepDamage(Entity<SurgeryTargetComponent> ent, ref SurgeryStepDamageEvent args) =>
         SetDamage(args.Body, args.Damage, args.PartMultiplier, args.User, args.Part);
 
+    // Begin DeltaV: surgery cross contamination
+    private FixedPoint2 Dirtiness(EntityUid entity)
+    {
+        var comp = EnsureComp<SurgeryDirtinessComponent>(entity);
+        return comp.Dirtiness;
+    }
+
+    private HashSet<string> CrossContaminants(EntityUid entity)
+    {
+        var comp = EnsureComp<SurgeryCrossContaminationComponent>(entity);
+        return comp.DNAs;
+    }
+
+    public FixedPoint2 TotalDirtiness(EntityUid user, List<EntityUid> tools, Entity<DnaComponent, SurgeryContaminableComponent> target)
+    {
+        var total = FixedPoint2.Zero;
+        var dnas = new HashSet<string>();
+
+        if (TryComp<BorgChassisComponent>(user, out var borg))
+        {
+            total += Dirtiness(user);
+            dnas.UnionWith(CrossContaminants(user));
+        }
+        else
+        {
+            if (_inventory.TryGetSlotEntity(user, "gloves", out var glovesEntity))
+            {
+                total += Dirtiness(glovesEntity.Value);
+                dnas.UnionWith(CrossContaminants(glovesEntity.Value));
+            }
+
+            foreach (var tool in tools)
+            {
+                total += Dirtiness(tool);
+                dnas.UnionWith(CrossContaminants(tool));
+            }
+        }
```

Comment:
replace this with a new event handled on BorgChassisComponent to prevent normal dirtiness checks

Reply:
i don't see a case where you'd want an event to conditionally cancel this, so i'm probably going to use a marker component instead